lib/parser.mly

```diff
@@ -1,599 +0,0 @@
-    {  grammar_logger "fun_app" ; FunApp (id, args) }
-  | TARGET LPAREN RPAREN
-    { grammar_logger "target_read" ; GetTarget }
-  | GETLP LPAREN RPAREN
-    { grammar_logger "get_lp" ; GetLP } (* deprecated *)
-  | id=identifier LPAREN e=expression BAR args=separated_list(COMMA, expression)
-    RPAREN
-    {  grammar_logger "conditional_dist_app" ; CondFunApp (id, e :: args) }
-  | LPAREN e=expression RPAREN
-    { grammar_logger "extra_paren" ; Paren e }
-
-%inline prefixOp:
-  | BANG
-    {   grammar_logger "prefix_bang" ; Not }
-  | MINUS
-    {  grammar_logger "prefix_minus" ; UMinus }
-  | PLUS
-    {   grammar_logger "prefix_plus" ; UPlus }
-
-%inline postfixOp:
-  | TRANSPOSE
-    {  grammar_logger "postfix_transpose" ; Transpose }
-
-%inline infixOp:
-  | a=arithmeticInfixOp
-    {   grammar_logger "infix_arithmetic" ; a }
-  | l=logicalInfixOp
-    {  grammar_logger "infix_logical" ; l }
-
-%inline arithmeticInfixOp:
-  | PLUS
-    {  grammar_logger "infix_plus" ; Plus }
-  | MINUS
-    {   grammar_logger "infix_minus" ; Minus }
-  | TIMES
-    {  grammar_logger "infix_times" ; Times }
-  | DIVIDE
-    {  grammar_logger "infix_divide" ; Divide }
-  | MODULO
-    {  grammar_logger "infix_modulo" ; Modulo }
-  | LDIVIDE
-    {  grammar_logger "infix_ldivide" ; LDivide }
-  | ELTTIMES
-    {  grammar_logger "infix_elttimes" ; EltTimes }
-  | ELTDIVIDE
-    {   grammar_logger "infix_eltdivide" ; EltDivide }
-  | HAT
-    {  grammar_logger "infix_hat" ; Exp }
-
-%inline logicalInfixOp:
-  | OR
-    {   grammar_logger "infix_or" ; Or }
-  | AND
-    {   grammar_logger "infix_and" ; And }
-  | EQUALS
-    {   grammar_logger "infix_equals" ; Equals }
-  | NEQUALS
-    {   grammar_logger "infix_nequals" ; NEquals}
-  | LABRACK
-    {   grammar_logger "infix_less" ; Less }
-  | LEQ
-    {   grammar_logger "infix_leq" ; Leq }
-  | RABRACK
-    {   grammar_logger "infix_greater" ; Greater }
-  | GEQ
-    {   grammar_logger "infix_geq" ; Geq }
-
-indexes:
-  | (* nothing *)
-    {   grammar_logger "index_nothing" ; [All] }
-  | COLON
-    {   grammar_logger "index_all" ; [All] }
-  | e=expression
-    {  grammar_logger "index_single" ; [Single e] }
-  | e=expression COLON
-    {  grammar_logger "index_upper" ; [Upfrom e] }
-  | COLON e=expression
-    {   grammar_logger "index_lower" ; [Downfrom e] }
-  | e1=expression COLON e2=expression
-    {  grammar_logger "index_twosided" ; [Between (e1, e2)] }
-  | i1=indexes COMMA i2=indexes
-    {  grammar_logger "indexes" ; i1 @ i2 }
-
-printables:
-  | e=expression
-    {  grammar_logger "printable expression" ; [PExpr e] }
-  | s=string_literal
-    {  grammar_logger "printable string" ; [PString s] }
-  | p1=printables COMMA p2=printables
-    { grammar_logger "printables" ; p1 @ p2 }
-
-(* L-values *)
-lhs:
-  | id=identifier
-    {  grammar_logger "lhs_identifier" ; (id, []) }
-  | l=lhs LBRACK id=indexes RBRACK
-    {  grammar_logger "lhs_index" ; (fst l, (snd l)@id) }
```

Comment:
Can you unpack in the binding? e.g.
`| (fstl, sndl) = lhs LBRACK id=indexes RBRACK`

Reply:
Sadly, no. Menhir does not let you do that.

Reply:
Boo, haha. Oh well, it does much else for us.